test/FunctionalTests/Infrastructure/ApplicationTestFixture.cs

```diff
@@ -67,56 +61,29 @@ public static DeploymentParameters GetDeploymentParameters(string applicationPat
             return deploymentParameters;
         }
 
-        public virtual ILoggerFactory CreateLoggerFactory()
-        {
-            return new LoggerFactory().AddConsole();
-        }
-
         public void Dispose()
         {
-        }
-
-        private static void TryDeleteDirectory(string directory)
-        {
-            try
+            if (_deploymentTask?.Status == TaskStatus.RanToCompletion)
             {
-                Directory.Delete(directory, recursive: true);
+                _deploymentTask.Result.HttpClient?.Dispose();
             }
-            catch (IOException)
-            {
-                // Ignore delete failures.
-            }
-        }
 
-        public async Task<Deployment> CreateDeploymentAsync(RuntimeFlavor flavor)
-        {
-            var deploymentParameters = GetDeploymentParameters(flavor);
-            var deployer = ApplicationDeployerFactory.Create(deploymentParameters, LoggerFactory);
-            var deploymentResult = await deployer.DeployAsync();
-
-            return new Deployment(deployer, deploymentResult);
+            _deployer?.Dispose();
         }
 
-        public class Deployment : IDisposable
+        public async Task<DeploymentResult> CreateDeploymentAsync(ILoggerFactory loggerFactory)
         {
-            public Deployment(IApplicationDeployer deployer, DeploymentResult deploymentResult)
+            lock (_deploymentLock)
             {
-                Deployer = deployer;
-                DeploymentResult = deploymentResult;
-                HttpClient = deploymentResult.HttpClient;
+                if (_deploymentTask == null)
+                {
+                    var deploymentParameters = GetDeploymentParameters();
```

Comment:
Basically this ensures that only one deployment happens per scenario. Saves about a minute and a half in tests where we  perform multiple things per application. I'm going to see if we can consolidate tests further.